Can't install jruby-head on 14.04 based systems (system maven is too outdated)

Hi everyone,

(most likely related to #3457)

I just tried to update my jruby-head like I used to so after doing `rvm get head` I did:

```
tobi@airship ~ $ rvm -v
rvm 1.26.11 (master) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]
tobi@airship ~ $ rvm reinstall jruby-head
Checking requirements for mint.
Requirements installation successful.
$JAVA_HOME was empty, setting up JAVA_HOME=/usr/lib/jvm/java-7-openjdk-amd64, if it fails try setting JAVA_HOME to something sane and try again.
Cloning from git://github.com/jruby/jruby.git, this may take a while depending on your connection.
HEAD is now at e3f84a0 do close jrubyclassloader on ScriptingContainer#terminate only
From git://github.com/jruby/jruby
 * branch            master     -> FETCH_HEAD
Already up-to-date.
git checkout master
Copying from repo to src path...
jruby-head - #mvn....
Error running 'mvn',
showing last 15 lines of /home/tobi/.rvm/log/1437426880_jruby-head/mvn.log
[INFO] JRuby Lib Setup ................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.596s
[INFO] Finished at: Mon Jul 20 23:18:38 CEST 2015
[INFO] Final Memory: 10M/303M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.0:enforce (default) on project jruby-parent: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Going a bit up in the log file one can see:

```
[INFO] --- maven-enforcer-plugin:1.0:enforce (default) @ jruby-parent ---
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
Detected Maven Version: 3.0.5 is not in the allowed range [3.3.0,).
```

Maven 3.0.5 is the newest thing we got in the Ubuntu 14.04 repositories. So people would need to install a newer Maven version - it'd be great if there would be a way around this. Call me lazy, but especially since autodeps I hoped rvm would handle this for me :) At the same time, I also think that 14.04 based systems are common enough so that it'd be great to work out of the box. (I'm running Linux Mint which is based in 14.04).

```
tobi@airship ~ $ uname -a
Linux airship 3.16.0-38-generic #52~14.04.1-Ubuntu SMP Fri May 8 09:43:57 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Cheers and thanks,
Tobi

--- a/scripts/functions/manage/jruby
+++ b/scripts/functions/manage/jruby
@@ -46,12 +46,14 @@ about not returning non zero status in case of errors."
 
 jruby_install_post17()
 {
-  __rvm_log_command "mvn" "$rvm_ruby_string - #mvn" mvn || return $?
+  \typeset mvnw="${rvm_src_path}/$rvm_ruby_string/mvnw"
+
+  __rvm_log_command "mvnw" "$rvm_ruby_string - #$mvnw" "$mvnw" || return $?
   if
     (( ${#rvm_configure_flags[@]} ))
   then
-    __rvm_log_command "mvn_flags" "$rvm_ruby_string - #mvn ${rvm_configure_flags[*]}" \
-      mvn "${rvm_configure_flags[@]}" ||
+    __rvm_log_command "mvn_flags" "$rvm_ruby_string - #$mvnw ${rvm_configure_flags[*]}" \
+      $mvnw "${rvm_configure_flags[@]}" ||
       return $?
   fi
 }

